docs/model_server_c_api.md

```diff
@@ -0,0 +1,45 @@
+# OpenVINO Model Server C-API (preview feature) {#ovms_docs_c_api}
+
+## Introduction
+
+This document describes OpenVINO Model Server (OVMS) C API that allows OVMS to be linked into C/C++ applications. With exceptions listed below all capabilities of OVMS are included in the shared library.
+
+## Preview limitations
+* Launching server in single model mode is not supported. You must use configuration file.
+* DAG pipelines cannot be used directly through C API. Inferences can be scheduled only by gRPC/HTTP endpoints.
+* There is no support for native file format (jpg/png) through C API.
+* There is no server live, server ready, model ready, model metadata, metrics endpoints exposed through C API.
+* Inference scheduled through C API are not accounted for in metrics.
+* There is no way to turn gRPC service off, HTTP service can be enabled.
+* There is no asynchronous inference.
+* There is support for stateful models.
+
+## API Description
+
+Server functionalities are encapsulated in shared library built from OVMS source. To include OVMS you need to link this library with your application and use C API defined in [header file](../src/ovms.h). To run OVMS you need to spawn process that will keep OVMS alive. Then you can schedule inference both directly from app using C API and gRPC/HTTP endpoints.
+
+### Server configuration
+
+To start OVMS you need to create `OVMS_Server` object using `OVMS_ServerNew`, with set of `OVMS_ServerSettings` and `OVMS_ModelsSettings` that describe how the server should be configured. Once the server is started using `OVMS_ServerStartFromConfigurationFile` you can schedule the inferences using `OVMS_Inference`. Example how to use OVMS with C/C++ application is [here](../demos/c_api_minimal_app/README.md).
+
+### Error handling
+Most of OVMS C API functions return `OVMS_Status` object pointer indicating the success or failure. Success is indicated by nullptr. Failure is indicated by returning `OVMS_Status` object. The status code can be extracted using `OVMS_StatusGetCode` function and the details of error can be retrieved using `OVMS_StatusGetDetails` function.
```

Comment:
```suggestion
Most of OVMS C API functions return `OVMS_Status` object pointer indicating the success or failure. Success is indicated by `NULL`. Failure is indicated by returning `OVMS_Status` object. The status code can be extracted using `OVMS_StatusGetCode` function and the details of error can be retrieved using `OVMS_StatusGetDetails` function.
```
Since this is C api and there is no nullptr, we should state NULL to be correct